service/controller/v12/resource/instance/template/main.json

```diff
@@ -106,7 +106,7 @@
               },
               {
                 "name":"DockerDisk",
-                "sizeGB":50
+                "sizeGB":"[if(greater(parameters('masterNodes')[0].dockerVolumeSizeGB, 0), parameters('masterNodes')[0].dockerVolumeSizeGB, 50)]"
```

Comment:
Would it make sense to have this conditional in the `newDeployment` method? Setting the defaults in Go code seems easier to read. What do you think?

Reply:
Default values and hardcoded values are all in the ARM templates. I feel putting this single default in the Go code would be choosing a different solution for an already solved problem, but I'm open to other opinions on this.

Reply:
@fiunchinho has a good point, but I think it's ok to have this in ARM template as a last catch-all parameter (so in all cases the template itself is functional with minimum configuration).

For legacy clusters (that all Azure clusters are) the defaulting is slightly distributed in several places (`api`, `cluster-service`, `kubernetesd` and `azure-operator`), but for newer style clusters with Cluster API types, the defaulting takes place in `api` now and later we most probably will introduce validation and mutating webhooks that can complement minimal CR YAMLs.